extension/src/router_channel.rs

```diff
@@ -485,10 +489,10 @@ impl RouterChannel {
           let chunk_data = chunk.as_ref().unwrap().data_ref().unwrap();
           encoder.write_all(chunk_data)?;
           encoder.flush()?;
-          let compressed_chunk = encoder.get_mut();
-          tx.send(Ok(Frame::data(compressed_chunk.clone().into())))
-            .await?;
-          compressed_chunk.clear();
+
+          let writer = encoder.get_mut();
+          let bytes = writer.get_mut().split().into();
+          tx.send(Ok(Frame::data(bytes))).await?;
```

Comment:
The `Bytes` type makes it cheap to slice off the front of the buffer and give it away to the frame data. Both `split` and `into` (convert from `BytesMut` to `Bytes`) are O(1).